woocommerce/woocommerce
# Add composer-bin-plugin

### All Submissions:


### Changes proposed in this Pull Request:

One of the caveats to using Composer when installing `bin` commands is that they become project dependencies. This has the unfortunate side-effect of creating version conflicts when two desired commands use incompatible versions of the same package. A nice Composer plugin exists called [composer-bin-plugin](https://github.com/bamarni/composer-bin-plugin) that is designed to solve this problem.

In this PR we put the bin commands in `bin/composer` and the package will symlink binaries from the subfolders into the root `vendor/bin` so that they can still be used as they currently are. I've gone ahead and moved PHPUnit, PHPCS, and the WP internationalization command into their own folders.

### How to test the changes in this Pull Request:

1. Run `composer install` and note that it will also install dependencies in the sub-folders.
2. Run `composer run-script phpcs` and note that it still works
3. Run `composer bin phpcs show` and note that it outputs information about the `bin/composer/phpcs/composer.json` file

You can create a new namespace with `composer bin new-namespace require xxxxx` as well to see how to add new commands. This should also allow us to move things like the PHPDoc generator into the Core repository for convenience.

### Other information:
